### PR TITLE
[COST-4118] - make all the workers attempt deletes before creating new files

### PR DESCRIPTION
**Pull request by lcouzens**
## Jira Ticket

[COST-4118](https://issues.redhat.com/browse/COST-4118)

## Description

This change will prevent any worker moving to create any parquet files until all previous files have been deleted! Essentially we tell all workers to delete any file in a certain path that DOES NOT match the processing manifest. 

## Testing

1. Checkout Branch
2. Restart Koku ideally with multiple workers
3. create ocp on aws/azure sources
4. ingest data that has multiple files per manifest
5. see in the logs that all deletions complete before creating new files.

## Notes
Running with multiple workers locally you should see something like this in the logs:

```
koku-koku-worker-4  | [2023-08-28 16:29:54,808] INFO c1390fbe-d1fc-4602-909b-a295288885dd 181 {'message': 'removed files from s3 bucket', 'tracing_id': '57088a4d-491d-415e-a9fb-436401482310', 'provider_uuid': 'cc24a00b-4b34-470c-8101-69779f83b0fe', 'provider_type': 'AWS-local', 'account': 'org1234567', 'bucket': 'koku-bucket', 'file_list': ['data/csv/org1234567/AWS/source=cc24a00b-4b34-470c-8101-69779f83b0fe/year=2023/month=08/2023-08-28_3.csv']}
koku-koku-worker-2  | [2023-08-28 16:29:54,854] INFO c1390fbe-d1fc-4602-909b-a295288885dd 313 {'message': 'removed files from s3 bucket', 'tracing_id': '57088a4d-491d-415e-a9fb-436401482310', 'provider_uuid': 'cc24a00b-4b34-470c-8101-69779f83b0fe', 'provider_type': 'AWS-local', 'account': 'org1234567', 'bucket': 'koku-bucket', 'file_list': ['data/csv/org1234567/AWS/source=cc24a00b-4b34-470c-8101-69779f83b0fe/year=2023/month=08/2023-08-28_3.csv']}
koku-koku-worker-3  | [2023-08-28 16:29:54,870] INFO c1390fbe-d1fc-4602-909b-a295288885dd 117 {'message': 'removed files from s3 bucket', 'tracing_id': '57088a4d-491d-415e-a9fb-436401482310', 'provider_uuid': 'cc24a00b-4b34-470c-8101-69779f83b0fe', 'provider_type': 'AWS-local', 'account': 'org1234567', 'bucket': 'koku-bucket', 'file_list': ['data/csv/org1234567/AWS/source=cc24a00b-4b34-470c-8101-69779f83b0fe/year=2023/month=08/2023-08-28_3.csv']}
koku-koku-worker-1  | [2023-08-28 16:29:54,874] INFO c1390fbe-d1fc-4602-909b-a295288885dd 470 {'message': 'removed files from s3 bucket', 'tracing_id': '57088a4d-491d-415e-a9fb-436401482310', 'provider_uuid': 'cc24a00b-4b34-470c-8101-69779f83b0fe', 'provider_type': 'AWS-local', 'account': 'org1234567', 'bucket': 'koku-bucket', 'file_list': ['data/csv/org1234567/AWS/source=cc24a00b-4b34-470c-8101-69779f83b0fe/year=2023/month=08/2023-08-28_3.csv']}
```

As you can see all workers downloading a file attempt to delete old files before processing.
Side note I condensed the output on the files list, so you will see more than this.
...

### CODE REVIEW

File: koku/masu/util/aws/common.py [hunk 1]
```diff
@@ -768,7 +769,7 @@ def delete_s3_objects(request_id, keys_to_delete, context) -> list[str]:
     return []
 
 
-def clear_s3_files(csv_s3_path, provider_uuid, start_date, context, request_id):
+def clear_s3_files(csv_s3_path, provider_uuid, start_date, metadata_key, metadata_value_check, context, request_id):
```
Comment (chambridge) [inline koku/masu/util/aws/common.py:772]: Is there currently a case where metadata_key is not "manifestid"? Just wondering if this is worth being a variable if you are just hardcoding it in a different location.

**Review by chambridge — APPROVED** [review]
Looks fine. One small caomment on the metadata_key

File: koku/masu/util/aws/common.py (hunk 1 above)
Comment (lcouzens) [reply koku/masu/util/aws/common.py:772]: So in this function no not right now, but OCP daily is using a different one somewhere else.

**Pull request merged**